README.md

```diff
@@ -139,6 +142,89 @@ To upgrade this resource from `4.0.6` to `4.2.7`:
    ```
 
+## Secure MongoDB Resource Connections using TLS
+
+You can configure the MongoDB Community Kubernetes Operator to use TLS certificates to encrypt traffic between:
+
+- MongoDB hosts in a replica set, and
+- Client applications and MongoDB deployments.
+
+### Prerequisites
+
+Before you secure MongoDB resource connections using TLS, you must:
+
+1. Using your own Certificate Authority (CA), create a PEM-encoded TLS certificate for each member of the Application Database’s replica set. The certificate must have one of the following:
+
+   - A wildcard `Common Name` that matches the domain name of all of the replica set members:
+
+     ```
+     *.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     ```
+   - The domain name for each of the replica set members as `Subject Alternative Names` (SAN):
+
+     ```
+     <metadata.name of the MongoDB resource>-0.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     <metadata.name of the MongoDB resource>-1.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     <metadata.name of the MongoDB resource>-2.<metadata.name of the MongoDB resource>-svc.<namespace>.svc.cluster.local
+     ```
+
+1. Create a Kubernetes ConfigMap that contains the certificate for the CA that signed your server certificate. For a certificate file named `ca.crt`:
+   ```
+   kubectl create configmap <tls-ca-configmap-name> --from-file=ca.crt --namespace <namespace>
+   ```
+
+1. Create a Kubernetes secret that contains the server certificate and key for the members of your replica set. For a server certificate named `server.crt` and key named `server.key`:
+   ```
+   kubectl create secret tls <tls-secret-name> --cert=server.crt --key=server.key --namespace <namespace>
+   ```
+
+### Procedure
+
+To secure connections to MongoDB resources using TLS:
+
+1. Add the following fields to the MongoDB resource definition:
+
+   - `security.tls.enabled`: Encrypts communications using TLS certificates between MongoDB hosts in a replica set and client applications and MongoDB deployments. Set to `true`.
+   - `security.tls.optional`: (**Optional**) Enables the members of the replica set to accept both TLS and non-TLS client connections. Equivalent to setting the MongoDB[`net.tls.mode`](https://docs.mongodb.com/manual/reference/configuration-options/#net.tls.mode) setting to `preferSSL`. If omitted, defaults to `false`.
+
+     ---
+     **NOTE**
+
+     When you enable TLS on an existing replica set deployment:
+
+     a. Set `security.tls.optional` to `true`.
+     b. Apply the configuration to Kubernetes.
+     c. Upgrade your existing clients to use TLS.
+     d. Set `security.tls.optional` to `false`.
+     e. Complete the remaining steps in the procedure.
+
+     ---
+   - `security.tls.certificateKeySecretRef.name`: Name of the Kubernetes secret that contains the server certificate and key that you created in the [prerequisites](#prerequisites-1).
```

Comment:
I think we should clarify that these settings go inside the top-level `spec` setting on the resource definition: `spec.security.tls.certificateKeySecretRef.name`.

Reply:
another good idea. updated.